kolena/_experimental/dataset/common.py

```diff
@@ -45,9 +45,11 @@ def validate_id_fields(id_fields: List[str], existing_id_fields: List[str] = Non
             )
 
 
-def _convert_non_int_str_to_str(df: pd.DataFrame) -> pd.DataFrame:
-    df = df.applymap(lambda x: str(x) if (not isinstance(x, (int, str))) else x)
-    return df
+def _convert_to_single_column(df):
+    # create a new dataframe with dictionary in each row
+    new_df = pd.DataFrame(df.apply(lambda x: x.to_dict(), axis=1))
```

Comment:
Nit: Series also support drop_duplicates. For our current usage, looks like we can just use `df[id_fields].apply(...).drop_duplcates()`

Alternatively, not sure if it saves a bit of memory or else
```
if <some_df>.duplicated().any():
  .....
```